datajoint/external.py

```diff
@@ -6,6 +6,203 @@
 from .declare import STORE_HASH_LENGTH, HASH_DATA_TYPE
 
 
+class ExternalFileHandler:
+
+    _handlers = {}  # handler mapping - statically initialized below.
+
+    def __init__(self, store, database):
+        self._store = store
+        self._database = database
+        self._spec = config[store]
+
+    @classmethod
+    def get_handler(cls, store, database):
+
+        if store not in config:
```

Comment:
Would catching the `KeyError` be a more efficient approach than explicit membership check?

Reply:
fundamentally disagree for the case of 'verifying key presence' especially when this is the part of the code directly responsible for validating configuration (aka verifying key presence) or membership checks are straightforward and dont require complex (exception causing error prone) logic 

a) 4 lines of code vs 2
b) exponentially? more places where bad indentation can bite you (especially in nested try blocks) 
d) is (try <> except KeyError <>) really more legible than 'if something not in container a else b'?

my take on exceptions is that they are just that - exceptional case to be handled via an alternate control structure not expressed clearly inline in the nominal path of the code.

also: this is not the same thing as running 50 duck-type tests for the same parameters on each stack frame pushed.